app/controllers/api/v1/book_lists_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Api::V1::BookListsController < ApplicationController
+  def index
+    render json: BookList.all
+  end
+
+  def create
+    book_list = BookList.create book_list_params
+    if book_list
+      render json: { message: "List created." }, status: :ok
+    else
+      render json: { message: "Error" }, status: :error
+    end
+  end
+
+  def add
+    book = Book.find(params[:book_id])
+    errors = []
+    params[:booklist_ids].each do |id|
+      booklist = BookList.find(id)
+      if booklist.books.include?(book)
+        errors << booklist.name
```

Comment:
I think it makes more sense to create the name of the error here on the server rather than the client. That way you can just print all the errors in javascript and you can have different types of errors